### PR TITLE
Issue3165  Term Entry: setLenient() method

### LINKED ISSUE #3165: [Term Entry] Java Calendar setLenient()
### Reason/inspiration (optional)

Please add the `setLenient()` method as a new term entry for the [Calendar](https://www.codecademy.com/resources/docs/java/calendar) class. Provide the entry in a markdown file and place under the file path `content/java/concepts/calendar/terms/setLenient/setLenient.md`

The entry should have the following:

–  An introductory paragraph about what the method is and does.
–  A syntax section with a pseudo code block with an explanation of it.
–  An example showing the method in use.

Please refer to the [term entry templates](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by ChiduAnush**
### Description

New Term Entry for ava Calendar setLenient(). Closes #3165 

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/calendar/terms/setLenient/setLenient.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Date'
+  - 'Methods'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The `**.setLenient()**` method of the `calender` class is used to set the lineacy of the interpretation of date and time. i.e., it is used to specify whether the internpretation of date and time should be lenient or not.
```
Comment (caupolicandiaz) [inline content/java/concepts/calendar/terms/setLenient/setLenient.md:14]:
```suggestion
The **`.setLenient()`** method of the `Calendar` class is used to set the leniency of the interpretation of date and time. In other words, it is used to specify whether the interpretation should be strict or not. 
```

File: content/java/concepts/calendar/terms/setLenient/setLenient.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+Tags:
+  - 'Date'
+  - 'Methods'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The `**.setLenient()**` method of the `calender` class is used to set the lineacy of the interpretation of date and time. i.e., it is used to specify whether the internpretation of date and time should be lenient or not. 
+The method takes one boolean value as a parameter.
```
Comment (caupolicandiaz) [inline content/java/concepts/calendar/terms/setLenient/setLenient.md:15]:
```suggestion
```

File: content/java/concepts/calendar/terms/setLenient/setLenient.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+> **Note:** When a `Calendar` is lenient, it allows date and time values that may not make sense in a real-world context. For example, `February 30th` or a time value of `25:70:90` would be accepted. When a `Calendar` is non-lenient, it enforces stricter rules, and attempting to set invalid date or time values will throw exceptions, such as `IllegalArgumentException`.
+
+## Syntax
+
+```pseudo
+calender.setLenient(boolean lenient)
```
Comment (caupolicandiaz) [inline content/java/concepts/calendar/terms/setLenient/setLenient.md:22]:
```suggestion
calender.setLenient(bool_flag)
```

File: content/java/concepts/calendar/terms/setLenient/setLenient.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+```
+Parameter: set `lenient` as false for more strict interpretation, that is, leniency is turned off. Set true for leniency to be turned on.
```
Comment (caupolicandiaz) [inline content/java/concepts/calendar/terms/setLenient/setLenient.md:24]:
```suggestion

This method has only one parameter, `bool_flag`, which should be set as `false` for a strict interpretation, that is, leniency is turned off. And set to `true` for leniency to be turned on.
```

File: content/java/concepts/calendar/terms/setLenient/setLenient.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+> **Note:** You can use the `.isLenient()` method to check if given calender instance is lenient or not.
```
Comment (caupolicandiaz) [inline content/java/concepts/calendar/terms/setLenient/setLenient.md:26]:
```suggestion
> **Note:** The `.isLenient()` method can be used to check if a given calendar is lenient or not.
```

File: content/java/concepts/calendar/terms/setLenient/setLenient.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+## Example 1
+
+This example shows what happens when you dont use .setLeient() method.
```
Comment (caupolicandiaz) [inline content/java/concepts/calendar/terms/setLenient/setLenient.md:30]:
```suggestion
This example demonstrates a use case for the  `.setLenient()` method.
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@ChiduAnush I've completed my initial review. Please update the entry, and tag me when you've finished making changes. Thanks

**Comment by ChiduAnush** [comment]
I did the recommended changes and did the git add and commit.
then I ran the `git push origin <my-branch-name>` command.

before that, I merged my branch and the Codecademy:main in my fork.

hopefully I did the right thing.
please check of the changes are now okay. thank you so much for your time

@caupolicandiaz

**Comment by ChiduAnush** [comment]
okay, as I was a little doubtful of the commits, 
I did the following commands, 
`git fetch upstream`
`git rebase upstream/main`
`git push -f origin <my branch name>`

I think it worked.
Please let me know if the changes are reflected properly.
@caupolicandiaz thank you again.

**Review by caupolicandiaz — APPROVED** [review]
@ChiduAnush I've made some additional edits directly. This is ready for a second review.

**Comment by ChiduAnush** [comment]
Thank you

**Comment by ChiduAnush** [comment]
I updated the branch using git fetch, rebase and push.
@caupolicandiaz should I see updating everyday or should I leave it?

**Comment by caupolicandiaz** [comment]
> I updated the branch using git fetch, rebase and push.
> @caupolicandiaz should I see updating everyday or should I leave it?

You don't have to worry about constantly updating. You can leave it as is.

**Review by SSwiniarski — APPROVED** [review]
@ChiduAnush Looks good to me.

**Comment by ChiduAnush** [comment]
great, thank you so much.
I can see that 2/6 checks have not passed.

what is this about? do I need to make any changes?
what are the next steps?

**Comment by ChiduAnush** [comment]
okay I can see the merged and closed purple notification. is my work done then? thank you too muchhh.
Can I see the changes on the website now?

**Pull request merged**